backend/api/routes/files.py

```diff
@@ -147,6 +151,52 @@ def validate_project_quota(file_size: int, project: Project):
         )
 
 
+def s3_file_key(project_id: UUID, path: str) -> str:
+    """Generate s3 file key."""
+    return f"{project_id}-{path}"
+
+
+def update_file_status_and_path(file: File, status: FileStatus, path: str):
+    """Update file's Status and Path."""
+    session = next(gen_session())
+    stmt = update(File).filter_by(id=file.id).values(status=status, path=path)
+    session.execute(stmt)
+    session.commit()
+
+
+def upload_file_to_s3(new_file: File):
+    """Update local file to S3 storage and update file status"""
+    s3_key = s3_file_key(new_file.project_id, new_file.hash)
+
+    if s3_storage.storage.has_object(s3_key):
+        update_file_status_and_path(new_file, FileStatus.S3, s3_key)
+        return
+
+    for i in range(constants.s3_max_tries):
+        try:
+            s3_storage.storage.upload_file(fpath=new_file.path, key=s3_key)
```

Comment:
You also must set an auto-delete date (matching the project's expiry) for every file you upload ; using `.set_object_autodelete_on()`

Reply:
FYI I have also set on the dev bucket (and we should as well on the prod before deploying) a 7d expiry meaning that every file will be automatically deleted after 7d. It's just a safety net